src/main/java/com/redhat/devtools/intellij/tektoncd/ui/component/StartTaskDialog.java

```diff
@@ -0,0 +1,459 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ * Red Hat, Inc.
+ ******************************************************************************/
+package com.redhat.devtools.intellij.tektoncd.ui.component;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.google.common.base.Strings;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.redhat.devtools.intellij.common.utils.YAMLHelper;
+import com.redhat.devtools.intellij.tektoncd.tkn.Resource;
+import com.redhat.devtools.intellij.tektoncd.tkn.component.field.Input;
+import com.redhat.devtools.intellij.tektoncd.tkn.component.field.Output;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.io.IOException;
+import java.util.List;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class StartTaskDialog extends DialogWrapper {
+    Logger logger = LoggerFactory.getLogger(StartTaskDialog.class);
+    private JPanel contentPane;
+    private JTextField inputParamValueTxt;
+    private JComboBox inputResourceValuesCB;
+    private JComboBox inputParamsCB;
+    private JComboBox outputsCB;
+    private JComboBox outputsResourcesCB;
+    private JTextArea previewTextArea;
+    private JButton previewRefreshBtn;
+    private JLabel outInfoMessage;
+    private JLabel inInfoMessage;
+    private JButton startTaskButton;
+    private JLabel errorLbl;
+    private JButton cancelButton;
+    private JComboBox inputResourcesCB;
+    private JPanel inputParamsPanel;
+    private JPanel inputResourcesPanel;
+    private JPanel outputsPanel;
+    private JLabel inputParamsLbl;
+    private JLabel inputParamValueLbl;
+    private JLabel outputsLbl;
+    private JLabel outputsResourceLbl;
+    private JLabel inputResourceValuesLbl;
+    private JLabel inputResourcesLbl;
+    private List<Input> inputs;
+    private List<Resource> resources;
+    private List<Output> outputs;
+
+    private String namespace;
+    private String taskName;
+    private List<String> args;
+
+    public StartTaskDialog(Component parent, String task, List<Resource> resources) {
+        super(null, parent, false, IdeModalityType.IDE);
+        try {
+            this.namespace = YAMLHelper.getStringValueFromYAML(task, new String[] {"metadata", "namespace"});
```

Comment:
Separation of concerns. The dialog implementation don't have to deal with YAML or JSON so it should be given an object representing the task. Maybe when we will switch to the Fabric8 Tekton API that would be more natural